code-gen-projects/rust/code-gen-demo/src/lib.rs

```diff
@@ -21,8 +21,8 @@ mod tests {
     }
 
     #[test_resources("../../input/good/struct_with_fields/**/*.ion")]
-    fn roundtrip_good_test_generated_code_structs_with_fields(file_name: &str) -> IonResult<()> {
-        let ion_string = fs::read_to_string(file_name)?;
+    fn roundtrip_good_test_generated_code_structs_with_fields(file_name: &str) -> SerdeResult<()> {
```

Comment:
Per offline discussion, `SerdeResult` will be a type that is generated alongside all the other code. This will guarantee that it's always available in codebases using codegen and does not require packaging a library per target language. It also means that we can evolve it more easily over time.